### ISSUE TITLE
Upgrade from C119 -> C120 on macOS x64 via 1.62.x - Nightly (Checking C120)

### ISSUE THREAD

**Issue by LaurenWags** [brave/brave-browser#34489]
As per process, QA runs through the following cases to ensure that the major chromium bump that's about to get merged into `master` doesn't regress the `Nightly` channel. Once the major chromium bump is merged into `master`, QA will run through a full manual pass on `Windows` & `Android`.

### Startup & Components

- [x] Verify that Brave is only contacting `*.brave.com` endpoints on first launch using either `Charles Proxy`, `Fiddler`, `Wireshark` or `LittleSnitch` (or a similar application)
  - [x] Verify that opening a NTP doesn't trigger any outbound connections related to widgets without user interaction
- [x] Remove the following component folders and ensure that they're being re-downloaded after restarting the browser:
  - [x] `afalakplffnnnlkncjhbmahjfjhmlkal`: `AutoplayWhitelist.dat`, `ExtensionWhitelist.dat`, `ReferrerWhitelist.json` and `Greaselion.json`
  - [x] `CertificateRevocation`
  - [x] `gccbbckogglekeggclmmekihdgdpdgoe`: (Sponsored New Tab Images)
  - [x] `gkboaolpopklhgplhaaiboijnklogmbc`: Brave Ad Block List Catalog
  - [x] `iodkpdagapdfkphljnddpjlldadblomo`: Brave Ad Block Updater
  - [x] `mfddibmblmbccpadfndgakiopmmhebop`: Brave Ad Block List Catalog
  - [ ] `oofiananboodjbbmdelgdommihjbkfag`: HTTPSE
  - [x] `Safe Browsing`
- [x] Restart the browser, load `brave://components`, wait for 8 mins and verify that no component shows any errors

**Note:** Always double check `brave://components` to make sure there's no errors/missing version numbers

### Upgrade

- [x] Make sure that data from the last version appears in the new version OK
- [x] Ensure that `brave://version` lists the expected Brave & Chromium versions
- [x] With data from the last version, verify that:
  - [x] Bookmarks on the bookmark toolbar and bookmark folders can be opened
  - [x] Cookies are preserved
  - [x] Installed extensions are retained and work correctly
  - [x] Opened tabs can be reloaded
  - [x] Stored passwords are preserved
  - [x] Sync chain created in previous version is retained
  - [x] Social media-blocking buttons changes are retained
  - [x] Custom filters under brave://settings/shields/filters are retained
  - [x] Custom lists under brave://settings/shields/filters are retained
  - [x] Rewards
    - [x] BAT balance is retained
    - [x] Auto-contribute list is retained
    - [x] Both Tips and Monthly Contributions are retained
    - [x] Panel transactions list is retained
    - [x] Changes to rewards settings are retained
    - [x] Ensure that Auto Contribute is not being enabled when upgrading to a new version if AC was disabled
  - [x] Ads
    - [x] Both `Estimated pending rewards` & `Ad notifications received this month` are retained
    - [x] Changes to ads settings are retained
    - [x] Ensure that ads are not being enabled when upgrading to a new version if they were disabled
    - [x] Ensure that ads are not disabled when upgrading to a new version if they were enabled

**Comment by LaurenWags** [comment]
Verification in progress with

```
Brave | 1.62.73 Chromium: 120.0.6099.28 (Official Build) nightly (x86_64)
-- | --
Revision | 522e9147d931744b1641084046c197caf7b341f0
OS | macOS Version 13.6.1 (Build 22G313)
```

**Comment by kjozwiak** [comment]
Verification **PASSED** on `macOS 14.1.1 x64 Sonoma` using the following build(s):

```
Brave | 1.62.74 Chromium: 120.0.6099.35 (Official Build) nightly (x86_64)
--- | ---
Revision | 070cd801c60b65bed66cfed719aeab87384ec9b8
OS | macOS Version 14.1.1 (Build 23B81)
```

<details>
<summary>C120 --> C120 on-demand upgrade check - PASSED</summary>

1. Install `C120` PR build (x64 builds available via https://bravesoftware.slack.com/archives/CU5UXM3EC/p1700676819867139?thread_ts=1700667812.477329&cid=CU5UXM3EC
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but close this tab prior to closing the browser)
4. Close `Brave`
5. Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nightly/appcast.xml`
6. Navigate to `brave://settings/help` and ensured that an update has been found/being applied

```
[4259:259:1126/231654.722912:INFO:sparkle_glue.mm(323)] brave update: did finish loading appcast
[4259:259:1126/231654.723726:INFO:sparkle_glue.mm(329)] brave update: did find valid update with AppcastItem(Date: Thu, 23 Nov 2023 02:29:50 +0000, Version: 1.62.74.0)
[4259:259:1126/231654.724393:INFO:sparkle_glue.mm(350)] brave update: willDownloadUpdate with AppcastItem(Date: Thu, 23 Nov 2023 02:29:50 +0000, Version: 1.62.74.0)
[4259:259:1126/231706.274935:INFO:sparkle_glue.mm(386)] brave update: will install update on quit with AppcastItem(Date: Thu, 23 Nov 2023 02:29:50 +0000, Version: 1.62.74.0)
```

7. Confirm you are prompted to "Relaunch" to get update

![Screenshot 2023-11-26 at 11 17 18 PM](https://github.com/brave/brave-browser/assets/2602313/0e1982d7-5acd-4309-bc6a-09b057879230)

Once relaunched, ensured the correct version was installed as per the following:

**`Example`** | **`Example`**
----------------|----------------
![Screenshot 2023-11-26 at 11 18 09 PM](https://github.com/brave/brave-browser/assets/2602313/e3e1b06d-e069-4779-b9a6-b7e5ed60453f) | ![Screenshot 2023-11-26 at 11 19 47 PM](https://github.com/brave/brave-browser/assets/2602313/184ca3aa-3b1e-471d-b58b-d6a3f08bb4de)

</details>


<details>
<summary>C119 --> C120 on-demand upgrade check - PASSED</summary>

1. Install `C119` build via https://github.com/brave/brave-browser/releases/tag/v1.62.73
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but close this tab prior to closing the browser)
4. Close `Brave`
5. Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/nightly/appcast.xml`
6. Navigate to `brave://settings/help` and ensured that an update has been found/being applied

```
[4584:259:1126/232534.744402:INFO:sparkle_glue.mm(323)] brave update: did finish loading appcast
[4584:259:1126/232534.745252:INFO:sparkle_glue.mm(329)] brave update: did find valid update with AppcastItem(Date: Thu, 23 Nov 2023 02:29:50 +0000, Version: 1.62.74.0)
[4584:259:1126/232534.746565:INFO:sparkle_glue.mm(350)] brave update: willDownloadUpdate with AppcastItem(Date: Thu, 23 Nov 2023 02:29:50 +0000, Version: 1.62.74.0)
[4584:259:1126/232539.385615:INFO:sparkle_glue.mm(350)] brave update: willDownloadUpdate with AppcastItem(Date: Thu, 23 Nov 2023 02:29:50 +0000, Version: 1.62.74.0)
[4584:259:1126/232627.460758:INFO:sparkle_glue.mm(386)] brave update: will install update on quit with AppcastItem(Date: Thu, 23 Nov 2023 02:29:50 +0000, Version: 1.62.74.0)
```

7. Confirm you are prompted to "Relaunch" to get update

![Screenshot 2023-11-26 at 11 28 04 PM](https://github.com/brave/brave-browser/assets/2602313/c0fd8845-e80e-4155-b6ee-ae66ca65fb60)

Once relaunched, ensured the correct version was installed as per the following:

**`Example`** | **`Example`**
----------------|----------------
![Screenshot 2023-11-26 at 11 28 33 PM](https://github.com/brave/brave-browser/assets/2602313/8c8e4f70-d17f-45b2-b9bc-25b934354d4d) | ![Screenshot 2023-11-26 at 11 28 40 PM](https://github.com/brave/brave-browser/assets/2602313/87004755-211a-4245-a83b-7104e41d0ff2)

</details>

<details>
<summary>C120 --> C120 background upgrade check - PASSED</summary>

1. Install `C120` PR build (x64 builds available via https://bravesoftware.slack.com/archives/CU5UXM3EC/p1700676819867139?thread_ts=1700667812.477329&cid=CU5UXM3EC
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but be sure to close this tab prior to closing the browser)
4. Close browser
5. Delete the following:
- `com.brave.Browser.nightly` folder from `~/Library/Caches`
- `com.brave.Browser.nightly.plist` file from `~/Library/Preferences`
- Alternatively you can probably advance your computer clock ~4 hours, I am fairly certain deleting at least one of the above files removes the timer check for background updates.
6. Restart your computer if you deleted files (if you just advanced computer clock this step isn't necessary)
7. After computer restart, Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nightly/appcast.xml`
8. DO NOT navigate to `brave://settings/help`
9. Watch the logs and see that an update is found (be sure you have all of the below before proceeding)

```
[946:259:1126/233643.979652:INFO:sparkle_glue.mm(323)] brave update: did finish loading appcast
[946:259:1126/233643.980604:INFO:sparkle_glue.mm(329)] brave update: did find valid update with AppcastItem(Date: Thu, 23 Nov 2023 02:29:50 +0000, Version: 1.62.74.0)
[946:259:1126/233643.981117:INFO:sparkle_glue.mm(350)] brave update: willDownloadUpdate with AppcastItem(Date: Thu, 23 Nov 2023 02:29:50 +0000, Version: 1.62.74.0)
[946:259:1126/233654.106152:INFO:sparkle_glue.mm(386)] brave update: will install update on quit with AppcastItem(Date: Thu, 23 Nov 2023 02:29:50 +0000, Version: 1.62.74.0)
```

10. Close and restart the browser via terminal with same command as step 7
11. Check version is updated under `brave://version`

![Screenshot 2023-11-26 at 11 40 53 PM (1)](https://github.com/brave/brave-browser/assets/2602313/b1408160-603e-4399-a146-7ced6ab17b7c)

</details>

<details>
<summary>C119 --> C120 background upgrade check - PASSED</summary>

1. Install `C119` build via https://github.com/brave/brave-browser/releases/tag/v1.62.73
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but be sure to close this tab prior to closing the browser)
4. Close browser
5. Delete the following:
- `com.brave.Browser.nightly` folder from `~/Library/Caches`
- `com.brave.Browser.nightly.plist` file from `~/Library/Preferences`
- Alternatively you can probably advance your computer clock ~4 hours, I am fairly certain deleting at least one of the above files removes the timer check for background updates.
6. Restart your computer if you deleted files (if you just advanced computer clock this step isn't necessary)
7. After computer restart, Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nightly/appcast.xml`
8. DO NOT navigate to `brave://settings/help`
9. Watch the logs and see that an update is found (be sure you have all of the below before proceeding)

```
[867:259:1126/234804.019625:INFO:sparkle_glue.mm(323)] brave update: did finish loading appcast
[867:259:1126/234804.020888:INFO:sparkle_glue.mm(329)] brave update: did find valid update with AppcastItem(Date: Thu, 23 Nov 2023 02:29:50 +0000, Version: 1.62.74.0)
[867:259:1126/234804.021585:INFO:sparkle_glue.mm(350)] brave update: willDownloadUpdate with AppcastItem(Date: Thu, 23 Nov 2023 02:29:50 +0000, Version: 1.62.74.0)
[867:259:1126/234809.794957:INFO:sparkle_glue.mm(350)] brave update: willDownloadUpdate with AppcastItem(Date: Thu, 23 Nov 2023 02:29:50 +0000, Version: 1.62.74.0)
[867:259:1126/234858.409179:INFO:sparkle_glue.mm(386)] brave update: will install update on quit with AppcastItem(Date: Thu, 23 Nov 2023 02:29:50 +0000, Version: 1.62.74.0)
[867:259:1126/234928.455996:INFO:sparkle_glue.mm(376)] brave update: will install update with AppcastItem(Date: Thu, 23 Nov 2023 02:29:50 +0000, Version: 1.62.74.0)
```

10. Close and restart the browser via terminal with same command as step 7
11. Check version is updated under `brave://version`

![Screenshot 2023-11-26 at 11 50 02 PM](https://github.com/brave/brave-browser/assets/2602313/2fb8a33a-c26d-4f26-b21b-9443bca92d83)
</details>
